lib/meandro/rules/unused_record_fields.ex

```diff
@@ -0,0 +1,158 @@
+defmodule Meandro.Rule.UnusedRecordFields do
+  @moduledoc """
+  Finds instances where a record has fields defined that are not being used.
+
+  This rule assumes that your code will never use the underlying tuple
+  structure of your records directly.
+  """
+
+  alias Meandro.Util
+
+  @behaviour Meandro.Rule
+
+  @impl Meandro.Rule
+  def analyze(files_and_asts, _options) do
+    for {file, ast} <- files_and_asts,
+        result <- analyze_module(file, ast) do
+      result
+    end
+  end
+
+  @impl Meandro.Rule
+  def is_ignored?({module, record, fields}, {module, record, fields}) do
+    true
+  end
+
+  def is_ignored?({_module, record, _fields}, record) do
+    true
+  end
+
+  def is_ignored?(_pattern, _ignore_spec) do
+    false
+  end
+
+  defp analyze_module(file, ast) do
+    {_, acc} = Macro.prewalk(ast, %{module: nil, records: []}, &collect_record_info/2)
+
+    for {module, name, _num_of_fields, unused_fields, line} <-
+          Enum.reverse(acc[:records]),
+        unused_fields != [] do
+      camel_name = name |> Atom.to_string() |> Macro.camelize()
+
+      %Meandro.Rule{
+        file: file,
+        line: line,
+        text:
+          "Record :#{name} (#{camel_name}) has the following unused fields in the module: #{inspect(unused_fields)}",
```

Comment:
I think it's better to emit one warning per field so they can be ignored *individually*.
The pattern will then be `{name, unused_field}` and people will be able to easily ignore the warning for just one field in the record.

Reply:
Or, alternatively, they'll be able to specify just the record name as the ignore pattern and ignore them all if they want to do so.

Reply:
I like the former suggestion more, albeit I feel like that would clutter the output a bit; all being said though, the `unused_struct_fields` does what you are suggesting, and consistency is the key :)

Reply:
And you usually shouldn't (**shouldn't**) have too many unused fields in the same record.